drivers/display/hd44780.zig

```diff
@@ -0,0 +1,276 @@
+const mdf = @import("../framework.zig");
+pub const lcd_delayus_callback = fn (delay: u32) void;
+
+pub const Lcd_commands = enum(u8) {
+    lcd_clear = 0x01,
+    lcd_reset = 0x02,
+    lcd_shift_cursor_left = 0x10,
+    lcd_shift_cursor_right = 0x14,
+    lcd_shift_display_left = 0x18,
+    lcd_shift_display_right = 0x1C,
+};
+
+//lcd "function set" enums
+const Function_set = enum(u8) {
+    lcd_line = 0x8,
+    lcd_bus = 0x10,
+    lcd_char = 0x4,
+};
+
+pub const Lines = enum(u8) { one, two };
+pub const Bussize = enum(u8) { bus4bits, bus8bits };
+pub const Charsize = enum(u8) { char5x8, char5x10 };
+
+//lcd "entry mode" enums
+const Entry_mode = enum(u8) { lcd_shift_mode = 0x1, lcd_shift_dir = 0x2 };
+pub const State = enum(u8) {
+    off,
+    on,
+};
+
+pub const ShiftDirection = enum(u8) {
+    dec,
+    inc,
+};
+
+//lcd "display control" enums
+const Display_control = enum(u8) { lcd_blink = 0x01, lcd_cursor = 0x02, lcd_display = 0x04 };
+
+pub const Config = struct {
+    lines: Lines = .two,
+    bus: Bussize = .bus4bits,
+    char_size: Charsize = .char5x8,
+    shift_direction: ShiftDirection = .inc,
+    display_shift: State = .off,
+    cursor: State = .off,
+    cursor_blink: State = .off,
+    display: State = .on,
+};
+pub const HD44780 = struct {
+    //internal config vars
+    function_set: u8,
+    entry_mode: u8,
+    display_control: u8,
+    enable_set: u8,
+    full_bus: bool,
+
+    //internal_callbacks
+    internal_delay: *const lcd_delayus_callback,
+    io_interface: mdf.base.Datagram_Device,
```

Comment:
This struct should follow the others, and have a `HD44780_Config` type which configures the `HD44780` struct. As you're using `Datagram_Device`, you should do something like
```zig
const HD44780_Config = struct {
  Datagram_Device: type = mdf.base.Datagram_Device,
};

pub fn HD44780(comptime config: HD44780_Config) type {
    return struct {
         …
         io_interface: config.Datagram_Device,
         …
    };
}
```
This way, you can either use the runtime dispatch variant with `HD44780(.{})` or you can replace the runtime dispatch with an arbitrary different dispatcher like a TestDevice or a concrete, non-vtable implementation